static/app/components/autoSelectText.tsx

```diff
@@ -1,69 +1,54 @@
-import {CSSProperties} from 'react';
+import {forwardRef, useImperativeHandle, useRef} from 'react';
 import * as React from 'react';
 import classNames from 'classnames';
 
-import {isRenderFunc} from 'app/utils/isRenderFunc';
 import {selectText} from 'app/utils/selectText';
 
-type ChildRenderProps = {
-  doSelect: () => void;
-  doMount: (el: HTMLElement) => void;
-};
-
-type ChildFunction = (props: ChildRenderProps) => React.ReactNode;
-
-type Props = {
-  /**
-   * Can be a `node` for a simple auto select div container.
-   * When children is a render function, it is passed 2 functions:
-   * - `doMount` - should be applied on parent element's `ref` whose
-   * children is the text to be copied
-   * - `doSelect` - selects text
-   */
-  children: React.ReactNode | ChildFunction;
+type Props = React.PropsWithChildren<{
   className?: string;
-  style?: CSSProperties;
-};
+  style?: React.CSSProperties;
+}>;
 
-class AutoSelectText extends React.Component<Props> {
-  private el: HTMLElement | undefined;
-
-  selectText = () => {
-    if (!this.el) {
-      return;
-    }
-
-    selectText(this.el);
-  };
+type AutoSelectHandle = {
+  selectText: () => void;
+};
 
-  handleMount = (el: HTMLElement) => {
-    this.el = el;
-  };
+const AutoSelectText: React.ForwardRefRenderFunction<AutoSelectHandle, Props> = (
+  {children, className, ...props},
+  forwardedRef
+) => {
+  const element = useRef<HTMLElement>();
 
-  render() {
-    const {children, className, ...props} = this.props;
+  // We need to expose a selectText method to parent components
+  // and need an imperitive ref handle.
+  useImperativeHandle(forwardedRef, () => ({
+    selectText: () => handleClick(),
+  }));
 
-    if (isRenderFunc<ChildFunction>(children)) {
-      return children({
-        doMount: this.handleMount,
-        doSelect: this.selectText,
-      });
+  function handleClick() {
+    if (!element.current) {
+      return;
     }
+    selectText(element.current);
+  }
 
-    // use an inner span here for the selection as otherwise the selectText
-    // function will create a range that includes the entire part of the
-    // div (including the div itself) which causes newlines to be selected
-    // in chrome.
-    return (
-      <div
-        {...props}
-        onClick={this.selectText}
-        className={classNames('auto-select-text', className)}
-      >
-        <span ref={this.handleMount}>{children}</span>
-      </div>
-    );
+  function handleMount(el: HTMLElement) {
```

Comment:
Since you're using a ref object, you can actually just pass `element` directly in like `ref={element}` and it will assign it.

Reply:
if you apply the suggestions I left, you can remove this function